addons/advanced_ballistics/functions/fnc_handleFired.sqf

```diff
@@ -48,19 +48,83 @@ if (GVAR(onlyActiveForLocalPlayers) && !(local _unit)) then {
     };
 };
 //if (!GVAR(vehicleGunnerEnabled) && !(_unit isKindOf "Man")) then { _abort = true; }; // We currently do not have firedEHs on vehicles
-if (GVAR(disabledInFullAutoMode) && getNumber(configFile >> "cfgWeapons" >> _weapon >> _mode >> "autoFire") == 1) then { _abort = true; };
+if (GVAR(disabledInFullAutoMode) && getNumber(configFile >> "CfgWeapons" >> _weapon >> _mode >> "autoFire") == 1) then { _abort = true; };
 
 if (_abort || !(GVAR(extensionAvailable))) exitWith {
-    [_bullet, getNumber(configFile >> "cfgAmmo" >> _ammo >> "airFriction")] call EFUNC(winddeflection,updateTrajectoryPFH);
+    [_bullet, getNumber(configFile >> "CfgAmmo" >> _ammo >> "airFriction")] call EFUNC(winddeflection,updateTrajectoryPFH);
 };
 
-_airFriction = getNumber(configFile >> "cfgAmmo" >> _ammo >> "airFriction");
+_AmmoCacheIndex = GVAR(AmmoCacheIndex) find _ammo;
+if (_AmmoCacheIndex == -1) then {
```

Comment:
I may be best to change how this is cached; currently you have to make a find for each shot fired, over a potentially large list of projectile types (well, I'm not sure how big that array could get).

An alternative would be to cache that data in global variables. E.g.;

```_WeaponCacheEntry